nkms/policy/models.py

```diff
@@ -0,0 +1,107 @@
+from nkms.crypto.hash import content_hash
+from nkms.crypto.pre.keygen import generate_re_encryption_keys
+from nkms.policy.constants import UNKNOWN_KFRAG
+
+
+class PolicyGroup(object):
+    """
+    The terms and conditions by which Alice shares with Bob.
+    """
+
+    def __init__(self, policies=None):
+        self.policies = policies or []
+
+    @staticmethod
+    def craft(keychain_alice: "KeyChain",
```

Comment:
Two questions here:

* Why `craft` and not `make`, for example? (or some commonly used word which you can guess without introspecting)
* Why staticmethod rather than, say, classmethod?

Reply:
I did some thinking about this class last night that I think obsoletes both of these questions: I think what we actually want here is a `PolicyManager` class to generate and manager `PolicyGroup` and `Policy` instances (ie, for "table-level" logic).

But to answer your questions:

1) I liked `craft` because this method has to be called by Alice, who is the one who originally crafts the policy.  
2) Yeah, a `@classmethod` is probably better long-term here, but I sometimes use `@staticmethod` at first before I'm 100% sure that the method will remain part of the class.